dgraphtest/image.go

```diff
@@ -16,6 +16,155 @@
 
 package dgraphtest
 
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/pkg/errors"
+)
+
+var (
+	homePath = os.Getenv("HOME") + "/dgraph"
+	repoDir  = homePath + "/binaries"
+)
+
+const (
+	relativeDir   = "binaries"
+	dgraphRepoUrl = "https://github.com/dgraph-io/dgraph.git"
+)
+
 func (c *LocalCluster) dgraphImage() string {
 	return "dgraph/dgraph:local"
 }
+
+func (c *LocalCluster) setupBinary(version string, logger Logger) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(time.Minute*10))
+	defer cancel()
+	absPath, err := filepath.Abs(repoDir)
+	if err != nil {
+		return errors.Wrap(err, "error while getting absolute path")
+	}
+	repo, err := git.PlainOpen(absPath)
+	if err != nil && err == git.ErrRepositoryNotExists {
+		logger.Logf("cloning repo")
+		repo, err = git.PlainCloneContext(ctx, absPath, false, &git.CloneOptions{
+			URL: dgraphRepoUrl,
+		})
+		if err != nil {
+			return errors.Wrap(err, "error while checking out dgraph git repo")
+
+		}
+	} else if err != nil {
+		return errors.Wrap(err, "error while opening git repo")
+	} else {
+		if err := repo.Fetch(&git.FetchOptions{}); err != nil && err != git.NoErrAlreadyUpToDate {
+			return errors.Wrap(err, "error while fetching git repo")
+		}
+	}
+	var hash plumbing.Hash
+	if strings.Contains(version, "v") {
```

Comment:
don't rely on this either. Just try both, first tag and if that doesn't work, then hash.